torch/_inductor/codegen/cpp.py

```diff
@@ -3542,6 +3639,11 @@ def _why_fuse_nodes(self, node1, node2) -> Optional[ReasonFusedNodes]:
             return ReasonFusedNodes.COMPATIBLE_REDUCTION
         if self._can_fuse_nodes_with_compatible_ranges(node1, node2):
             return ReasonFusedNodes.COMPATIBLE_RANGES_NO_REDUCTION
+        if self._get_outer_loop_fusion_status() and (
+            vars1 + reduce1 == vars2 and reduce2 == ()
```

Comment:
nb. I wrote this following https://github.com/pytorch/pytorch/pull/121064#issuecomment-1980895900. This could be useful if you have something of the form `x = torch.randn(m, n); x1 = x.sum(-1); x[:, idx] + x1`. That being said, it may require a reasonable extension to the fusion logic, so it may not be too high prio.

Reply:
Noted. Looks like another following up after this PR.